### ISSUE TITLE
Sentient carp ghost roles need to be cut down

### ISSUE THREAD

**Issue by UbaserB** [space-wizards/space-station-14#31424]
## Description

The list gets far too long and troublesome in the ghost role menu when the dragon has a lot of carps, so the fix to that would be adding a *chance* for each carp to be sentient when spawned, or just not sentient at all (which has its own downsides, but is consistent). If the former, that makes sentient carp ghost roles a desired role which is also better imo.

**Comment by coolsurf6** [comment]
I think the first few roles spawned from a portal should always be sentient, giving people a chance expecially early on to join the Dragon. 

Maybe after the portal has fully charged, only a certain percentage (or none) if there is over X (10... 20...?) carp roles.

**Comment by Ilya246** [comment]
or just have portals keep track of the amount of sentient carp this portal spawned that are currently alive and only have at most like 6 per portal

**Comment by Errant-4** [comment]
This proposed solution treats a symptom rather than the cause. Rather, the ghostrole spawn list should be able to collapse large groups or something like that

**Comment by UbaserB** [comment]
> This proposed solution treats a symptom rather than the cause. Rather, the ghostrole spawn list should be able to collapse large groups or something like that

that is another fix, and something that is also needed eventually but what i proposed is still valid imo

**Comment by Anzarot121** [comment]
The problem is that if there are no ghostroles of sentient carp, the rifts will become essentially useless, because the AI of the carp is turned off when it hits the space tile.

**Comment by UbaserB** [comment]
> The problem is that if there are no ghostroles of sentient carp, the rifts will become essentially useless, because the AI of the carp is turned off when it hits the space tile.

thats an unrelated bug and we shouldn’t balance it around that cause when it gets fixed the balance will get out of whack

**Comment by Anzarot121** [comment]
> thats an unrelated bug and we shouldn’t balance it around that cause when it gets fixed the balance will get out of whack

It's related. Removing the carp roles will nerf the dragon. Because without players, any number of carp guarding the portal will be killed by a player with PKA.

**Comment by JIPDawg** [comment]
> This proposed solution treats a symptom rather than the cause. Rather, the ghostrole spawn list should be able to collapse large groups or something like that
> 
> (edit: Ironically, I kinda got that backwards since spawn list being bloated IS the symptom and dragon spawning so many carps IS the cause, or rather, A cause. But anyway)

I've had this idea in my head, being collapsing the groups of ghost roles. Was mainly waiting on some of the UI changes to be merged but not sure when that'll be.

**Comment by EthanQix** [comment]
If the ghost role list is full of carps, this means the Dragon has likely taken over the round and there are plenty of ghosts to play the carps. Reducing the number of carp roles will only end up with longer raffle queues and frustrated ghosts.

Bottom line, this is a UI issue. You don't fix a UI issue by changing mechanics, but by changing the UI instead. As mentioned above, collapsing identical ghost role lists would fix the problem for the carps, but also the mice or any other multiple roles.

**Comment by ArtisticRoomba** [comment]
Ghost role groups are now collapsible, as what has been proposed in this issue. However there is still the original proposal of making the carps have a chance at sentience rather than having them all being sentient and available as ghost roles.

You are free to close this issue yourself or keep it open.

**Comment by FungiFellow** [comment]
I might go PR a 70-80% chance of sentience for the dragon carp, so you still have a decent chunk of AI Carp while also having sentient carp

**Comment by Entvari** [comment]
> I might go PR a 70-80% chance of sentience for the dragon carp, so you still have a decent chunk of AI Carp while also having sentient carp

This issue was made because the flooded they ghostrole menu, the ghostrole menu is now collapsible so this is no longer an issue.
Less sentient carps has literally no benefit at this point.

**Comment by Errant-4** [comment]
PR has been superseded by #32717
